src/background_music/Loudness.h

```diff
@@ -0,0 +1,194 @@
+#ifndef AUDACIOUS_PLUGINS_BGM_LOUDNESS_H
+#define AUDACIOUS_PLUGINS_BGM_LOUDNESS_H
+/*
+ * Background music (equal loudness) Plugin for Audacious
+ * Copyright 2023 Michel Fleur
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions, and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the following disclaimer in the documentation
+ *    provided with the distribution.
+ *
+ * This software is provided "as is" and without any warranty, express or
+ * implied. In no event shall the authors be liable for any damages arising from
+ * the use of this software.
+ */
+#include "Integrator.h"
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <libaudcore/ringbuf.h>
+
+/**
+ * Tools to detect perceived loudness.
+ */
+struct Loudness
+{
+    static constexpr float perception_center_seconds = 0.400;
+    static constexpr float perception_fast_seconds = 0.001;
+    static constexpr float perception_peak_seconds = 0.00001;
+    static constexpr float perception_weight_power = 0.25;
+    static constexpr float max_prediction = 0.03;
+    static constexpr float window_percentage = 1.0;
+
+    struct Metrics
+    {
+        int window_samples = 1;
+        int latency_samples = 0;
+        float weight = 0;
+    };
+
+    static constexpr float get_weight(const float seconds)
+    {
+        const float relative = std::clamp(seconds, perception_peak_seconds,
+                                          perception_center_seconds) /
+                               perception_center_seconds;
+        return powf(relative, perception_weight_power);
+    }
+    static int get_samples(const float seconds, const int rate)
```

Comment:
Missing blank line

Reply:
Fixed part of [this commit](https://github.com/audacious-media-player/audacious-plugins/commit/2084df6f3f4bdcc024aa83000de3f1f319ba6d16)